fix php fatal error when no route match.

brettmc opened the pull request:
This adds checks that a route was matched before trying to use the RouteMatch object for further assertions. It also adds a new assertion, testNoRouteMatched(), so you can assert that no route was matched

brettmc committed:
fix php fatal error when no route match.
This adds checks that a route was matched before trying to use the RouteMatch object for further assertions. It also adds a new assertion, testNoRouteMatched(), so you can assert that no route was matched

brettmc commented:
Hi. I also wanted to mention, you could argue that with the negation assertions (eg assertNotMatchedRouteName()) you could argue that they should pass rather than fail if no route was matched.

brettmc committed:
phpcs fixes

Maks3w commented:
There are new 7 execution paths without test. Could you add tests for each one of the expected exceptions?

brettmc committed:
more unit tests for no route match.
adds tests for the various assertions that now fail with a phpunit exception when no route matches.

brettmc committed:
phpcs fixes

weierophinney commented on test/PHPUnit/Controller/AbstractControllerTestCaseTest.php:
```diff
@@ -248,6 +248,68 @@ public function testAssertNotMatchedRouteName()
+    public function testAssertNoMatchedRouteWithMatchedRoute()
+    {
+        $this->dispatch('/tests');
+        $this->setExpectedException('PHPUnit_Framework_ExpectationFailedException');
```
Each of these should also provide a message to match (i.e., `'No route matched'`), in part to make it more clear what sort of failure is expected.

I can add those on merge, however.
